packages/client/lib/sync/fullsync.ts

```diff
@@ -220,7 +220,7 @@ export class FullSynchronizer extends Synchronizer {
     }
 
     const knownBlocks = this.newBlocksKnownByPeer.get(peer.id) ?? []
-    if (knownBlocks?.filter((knownBlock) => knownBlock.hash.equals(blockHash))) {
+    if (knownBlocks?.filter((knownBlock) => knownBlock.hash.equals(blockHash)).length > 0) {
```

Comment:
This whole method seems a bit repetitive in the code, I would rewrite it like this:

```typescript
const knownBlocks = this.newBlocksKnownByPeer.get(peer.id) ?? []

if (knownBlocks.find((known) => known.hash.equals(blockHash))) {
  return true
}

knownBlocks.push({ hash: blockHash, added: Date.now() })
this.newBlocksKnownByPeer.set(peer.id, knownBlocks)
return false
```